src/common/exception/ErrorMessage.jl

```diff
@@ -0,0 +1,220 @@
+# assigned on the right side pair
+const error_messages = Dict([
+        Client_CLIENT_CLOSED=>( 1, "The client has been closed and no further operation is allowed."),
+        Client_SESSION_CLOSED=>( 2, "The session has been closed and no further operation is allowed."),
+        Client_TRANSACTION_CLOSED=>( 3, "The transaction has been closed and no further operation is allowed."),
+        Client_UNABLE_TO_CONNECT=>(4, "Unable to connect to Grakn server."),
+        Client_NEGATIVE_VALUE_NOT_ALLOWED=>(5, "Value cannot be less than 1, was: '%d'."),
+        Client_MISSING_DB_NAME=>(6,  "Database name cannot be null."),
+        Client_DB_DOES_NOT_EXIST=>(7,  "The database '%s' does not exist."),
+        Client_MISSING_RESPONSE=>(8,  "Unexpected empty response for request ID '%s'."),
+        Client_UNKNOWN_REQUEST_ID=>(9,  "Received a response with unknown request id '%s'."),
+        Client_CLUSTER_NO_PRIMARY_REPLICA_YET=>(10,  "No replica has been marked as the primary replica for latest known term '%d'."),
+        Client_CLUSTER_UNABLE_TO_CONNECT=>(11,  "Unable to connect to Grakn Cluster. Attempted connecting to the cluster members, but none are available: '%s'."),
+        Client_CLUSTER_REPLICA_NOT_PRIMARY=>(12,  "The replica is not the primary replica."),
+        Client_CLUSTER_ALL_NODES_FAILED=>(13,  "Attempted connecting to all cluster members, but the following errors occurred: \n%s"),
+    
+        ### Conept Error Section
+        Concept_INVALID_CONCEPT_CASTING=>(1, "Invalid concept conversion from '%s' to '%s'."),
+        Concept_MISSING_TRANSACTION=>(2,  "Transaction cannot be null."),
+        Concept_MISSING_IID=>(3,  "IID cannot be null or empty."),
+        Concept_MISSING_LABEL=>(4,  "Label cannot be null or empty."),
+        Concept_BAD_ENCODING=>(5,  "The encoding '%s' was not recognised."),
+        Concept_BAD_VALUE_TYPE=>(6,  "The value type '%s' was not recognised."),
+        Concept_BAD_ATTRIBUTE_VALUE=>(7,  "The attribute value '%s' was not recognised."),
+
+        # Query Error Section
+        Query_VARIABLE_DOES_NOT_EXIST=>(1,  "The variable '%s' does not exist."),
+        Query_NO_EXPLANATION=>(2,  "No explanation was found."),
+        Query_BAD_ANSWER_TYPE=>(3,  "The answer type '%s' was not recognised."),
+        Query_MISSING_ANSWER=>(4,  "The required field 'answer' of type '%s' was not set."),
+
+        #Internal Error Section
+        Internal_UNEXPECTED_INTERRUPTION =>(1, "Unexpected thread interruption!"),
+        Internal_ILLEGAL_STATE =>(2, "Illegal state has been reached!"),
+        Internal_ILLEGAL_ARGUMENT =>(3, "Illegal argument provided: '%s'"),
+        Internal_ILLEGAL_CAST =>(4, "Illegal casting operation to '%s'.")
+])    
+
+
+# Build the error massage according the section of errors represented by the 
+# abstract type of each section of errors
+function _buidl_error_messages(class::Type{T}) where {T<:AbstractClientError}
```

Comment:
Typo.. should be `build` not `buidl`. Same comments for functions below.

Reply:
I also don't understand what it's building... It's returning a single instance of the class but it's not passing any variable arguments for the string template. Perhaps this is still WIP?

It's nice start though! Let me think a little more about this design and come back with more suggestions later.

Reply:
Maybe it will be more clear if we implement the next file GraknClientException.jl. The usage of this will be more clear there.